### PR TITLE
fix(components): update type styles

### PR DESCRIPTION
**Pull request by asudoh**
Refs #1945.

#### Changelog

**Changed**

- Type styles of various components, details below.

#### Testing / Reviewing

@IBM/carbon-designers: It'd be great if you can review focusing (only) on type tokens of the following components:

* Feedback text of code snippet (Couldn't find a spec for that in #1888)
* Date picker (#1927 is closed as fixed but still found usages of deprecated type tokens)
* Helper text for various form components
* Modal (#1843 does not cover type tokens)
* Progress indicator (In "needs review" state)
* Radio button (#1926/#1962 do not cover type tokens)
* Search (In "needs review" state)
* Structured list (Just removed a usage of deprecated type token which seems not needed)
* Tabs (In "needs review" state)
* Tag (In "stable" state, but still found usages of deprecated type tokens)
* Text area (#1946 does not cover type tokens)
* Toggle (#1911 does not cover type tokens)

Thanks!

BTW did _not_ touch the following components:

* Dropdown (Covered by #1955)
* Number input (Covered by #1939)
* Pagination (Covered by #1957)
* Select (Covered by #1958)
* Tooltip (Covered by #1954)

### CODE REVIEW

File: src/components/search/_search.scss [hunk 1]
```diff
@@ -234,17 +234,17 @@
...
 
   .#{$prefix}--search--lg .#{$prefix}--search-input {
-    @include typescale('epsilon');
     height: rem(40px);
   }
 
   .#{$prefix}--search--xl .#{$prefix}--search-input {
-    @include typescale('epsilon');
+    @include type-style('body-short-02');
+    font-size: carbon--type-scale(4);
```
Comment (aagonzales) [inline src/components/search/_search.scss:247]: What is `font-size: carbon--type-scale(4);` for and how is it different than the type style?

**Review by aagonzales — APPROVED** [review]
Looks good! I just have to one question where theres a size and a type token

File: src/components/progress-indicator/_progress-indicator.scss [hunk 2]
```diff
@@ -266,8 +265,7 @@
     position: absolute;
     margin-left: $spacing-lg;
     margin-top: rem(28px);
-    @include font-size('12');
-    color: $text-02;
+    @include type-style('label-01') color: $text-02;
```
Comment (tw15egan) [inline src/components/progress-indicator/_progress-indicator.scss]: Should this be broken into two lines?

File: src/components/search/_search.scss (hunk 1 above)
Comment (tw15egan) [reply src/components/search/_search.scss:247]: I'm also unsure what this is used for

**Review by tw15egan — CHANGES_REQUESTED** [review]
Just a few small changes, otherwise good to go

File: src/components/search/_search.scss (hunk 1 above)
Comment (asudoh) [reply src/components/search/_search.scss:247]: Hi @aagonzales @tw15egan good question - There seems no migration path for `
typescale('epsilon')`, and there seems no `type-style()` corresponding to the font size for large search. That said, `font-size: carbon--type-scale(4)` is for ensuring the font size is the one of large search. Does it make sense to you...?

File: src/components/progress-indicator/_progress-indicator.scss (hunk 2 above)
Comment (asudoh) [reply src/components/progress-indicator/_progress-indicator.scss]: Great catch @tw15egan thanks! Fixed.

File: src/components/search/_search.scss (hunk 1 above)
Comment (aagonzales) [reply src/components/search/_search.scss:247]: Yes makes sense but why is `@include type-style('body-short-02');` still included then. Are you just overriding the size?

**Review by tw15egan — APPROVED** [review]

File: src/components/search/_search.scss (hunk 1 above)
Comment (asudoh) [reply src/components/search/_search.scss:247]: > Are you just overriding the size?

Yes, bigger font size, everything else is same as `type-style('body-short-02')`.

**Pull request merged**